flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonParserToRowDataConverters.java

```diff
@@ -266,7 +266,7 @@ private int convertToTime(JsonParser jsonNode) throws IOException {
         LocalTime localTime = parsedTime.query(TemporalQueries.localTime());
 
         // get number of milliseconds of the day
-        return localTime.toSecondOfDay() * 1000;
+        return (int) (localTime.toNanoOfDay() / 1000_000);
```

Comment:
If we change the way that the json format parses times, then the output will change for existing users, which has  implications to the logic of their processing. I suggest that this way of parsing be enabled by a new json format configuration option - so it can be knowingly adopted.